### DISCUSSION TITLE
Get all releases (singles) for a artist.

### DISCUSSION THREAD

**Discussion by bornsilly** [joalla/discogs_client#149] [Support] [answered]
Hello,

This might be a noob question but in the all the examples I cannot find anything if I want to see all the releases from a specific artist (and then preferably only the singles).  All I see in the examples is getting information from a release. 

Hope anybody doesn't mind sharing some code on how to get artist releases by artists ID. 

Thanks in advance

**Comment by prcutler** [comment, marked as answer]
This page in the docs might help:  https://python3-discogs-client.readthedocs.io/en/latest/fetching_data_repl.html#artist

You can get a list of releases by page, but I'm not sure how to get just singles.  Here's an example from the REPL and the docs, try plugging in a different artist ID and see what happens:

```py
>>> import discogs_client
>>> d = discogs_client.Client('TestREPL/0.1')
>>> artist = d.artist(956139)
>>> artist.releases.page(1)
[<Master 343357 "It Can't Be Forever / Best Of Every Year">, <Release 15939415 'The Way We Love'>, <Master 447055 'Thicker Than Water'>, <Release 22079767 'Footloose - Promo Radio'>, <Release 18991960 'The Normal Heart'>, <Master 90721 'Apollo 13 (Music From The Motion Picture)'>, <Master 1737668 'Forosoco'>, <Master 204895 'Destination Anywhere'>, <Release 12304381 'Sony Music Radio Sampler (April 1998)'>, <Release 30777201 'Sexcrimes (Wild Things) - Multimedia Press Kit'>, <Master 1808171 'Getting There'>, <Master 1289718 "Can't Complain">, <Master 1230194 "Come Together: A Night For John Lennon's Words And Music">, <Master 1636318 'Where The Truth Lies (Original Motion Picture Soundtrack)'>, <Release 9849590 'New Country Rock Vol. 11'>, <Release 15939415 'The Way We Love'>, <Master 3315730 'The Guardians Of The Galaxy Holiday Special (Original Soundtrack)'>, <Master 3470347 'The Ballad Of The Brothers '>, <Master 90721 'Apollo 13 (Music From The Motion Picture)'>, <Master 1230194 "Come Together: A Night For John Lennon's Words And Music">, <Release 11863658 'Guts Records Sampler 2006 赤裸裸'>, <Master 3315730 'The Guardians Of The Galaxy Holiday Special (Original Soundtrack)'>, <Master 1230194 "Come Together: A Night For John Lennon's Words And Music">]
```

One of the core devs might know more, but this might help get you started.

**Reply by JOJ0** [reply]
In the output above we see that

```
artist.releases.page(1)
```

returns a list of releases. Just loop through that and put conditions on what you are looking for. I'm sure there is a field in each release object that holds whether it's a single or other type of release.

Does that help already or you need more details?

**Reply by JOJ0** [reply]
this should be the release type: https://python3-discogs-client.readthedocs.io/en/latest/discogs_client.html#discogs_client.models.Release.formats

**Reply by bornsilly** [reply]
Thank you .. you are all very helpfull. I managed to indeed loop through it and get my what i need..

**Reply by bornsilly** [reply]
I can see all the data now, from all the artists. but how do you loop next through the individual release.  I can see it a MixedPaginedList and each in this list is a  discogs_client.models.Release but how do you loop through for example artists.releases.page(1) or just through  artists.releases?
